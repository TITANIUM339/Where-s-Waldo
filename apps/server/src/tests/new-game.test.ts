import express from "express";
import request from "supertest";
import { describe, expect, test } from "vitest";
import errorHandler from "../middleware/error-handler";
import gameRouter from "../routes/games/game/game";

const app = express();

app.use(gameRouter);

app.use(errorHandler);

describe("responds with data for a new game", () => {
    test("GET /1/new-game", async () => {
        const response = await request(app).get("/1/new-game");

        expect(response.body?.game).toMatchSnapshot();
        expect(response.body?.characters).toMatchSnapshot();
        expect(response.body?.token).toMatch(/^[\w-]+\.[\w-]+\.[\w-]+$/);
    });
});
