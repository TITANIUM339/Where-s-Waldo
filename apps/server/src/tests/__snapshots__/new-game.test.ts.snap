// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`responds with data for a new game > GET /1/new-game 1`] = `
{
  "characters": [
    {
      "id": 1,
      "imageURL": "https://static.wikia.nocookie.net/waldo/images/9/9d/Character.Waldo.jpg",
      "name": "Waldo",
    },
    {
      "id": 2,
      "imageURL": "https://static.wikia.nocookie.net/waldo/images/3/3e/Character.Wenda.jpg",
      "name": "Wenda",
    },
    {
      "id": 3,
      "imageURL": "https://static.wikia.nocookie.net/waldo/images/6/60/Character.Whitebeard.jpg",
      "name": "Wizard Whitebeard",
    },
  ],
  "description": "WHAT A SPECTACULAR SCENE THIS IS, WALDO-WATCHERS! AND WHAT AN EPIC COMMOTION PICTURE! I WONDER WHY THE TROJANS DIDN'T GUESS THE WOODEN HORSE WAS FULL OF GREEKS, AND HOW DID THEY GET IT THROUGH THE GATES OF TROY ANYWAY? I WOULDN'T LIKE TO BE IN THE TROJANS' SANDALS, IF THE COSTUME DEPARTMENT HAD GIVEN THEM ANY, THAT IS!",
  "id": 1,
  "imageHeight": 2227,
  "imageURL": "https://i.imgur.com/QLACwOG.jpeg",
  "imageWidth": 3536,
  "name": "HORSEPLAY IN TROY",
}
`;

exports[`responds with data for a new game > GET /1/new-game 2`] = `undefined`;
